samples/docdb-scaling/scaleupdown.py

```diff
@@ -0,0 +1,204 @@
+    parser.add_argument('--cluster-id',
+                        required=True,
+                        help='Amazon DocumentDB Cluster Identifer')
+    
+    parser.add_argument('--target-instance-type',
+                        required=True,
+                        type=str,
+                        help='target instance type to perform scale up/down operation on given cluster id')
+
+                        
+    parser.add_argument('--ignore-instance-check',
+                       required=False,
+                       action='store_false',
+                       help='ignore check for existing instances with target instance type (Optional)')
+    
+    args = parser.parse_args()
+
+    cluster_id=args.cluster_id
+    itarget=args.target_instance_type
+    monitor_deletion=False
+    target_instance_check=args.ignore_instance_check
+    valid_instance_classes=["db.r6g.large","db.r6g.xlarge","db.r6g.2xlarge","db.r6g.4xlarge","db.r6g.8xlarge","db.r6g.12xlarge","db.r6g.16xlarge","db.r5.large","db.r5.xlarge","db.r5.2xlarge","db.r5.4xlarge","db.r5.8xlarge","db.r5.12xlarge","db.r5.16xlarge","db.r5.24xlarge","db.t4g.medium","db.t3.medium"]
+   
+    if itarget not in valid_instance_classes:
+        LOGGER.error("Target instance type "+ itarget +" is invalid. Exiting..")
+        sys.exit(1)
+   
+    if target_instance_check and checkExistingInstanceClass(cluster_id=cluster_id,itarget=itarget):
+        LOGGER.error("Cluster already has one of the instance with target instance type.If you want to still continue,please run with --ignore-instance-check.Exiting..")
+        sys.exit(1)
+
+    if not isClusterHealthy(cluster_id):
+        LOGGER.error("Cluster is not healthy. Exiting..")
+        sys.exit(1)
+        return False 
+    
+    num_instances=getInstanceStatus(cluster_id)["num_instances"]
+    oinstances=list((getClusterInstances(cluster_id)).keys())
+    
+    if num_instances > 8:
+        LOGGER.error("Number of instances can not be more than 8 for this script.")
+        sys.exit(1)
+        return False
+    
+    ## Add instances 
+    LOGGER.info("Adding "+ str(len(oinstances))+  " instances with instance type " + itarget)
+    ninstances=addInstances(cluster_id=cluster_id,desired_size=itarget,count=num_instances)
+    LOGGER.info("checking  cluster health after instance addition" )
+    count=0
+    if checkClusterHealth(cluster_id=cluster_id):
+        LOGGER.info("Cluster is healthy post instance addition,performing failover")
+    else:
+        LOGGER.info("Cluster is not healthy post instance addition,existing")
+        sys.exit(1)
+    
+    failover_target=ninstances[0]
+    old_writer=getWriterInstance(cluster_id=cluster_id)
+    LOGGER.info("Failover target - " + failover_target)
+    
+    ## Perform failover 
+    perfromFailover(cluster_id=cluster_id,failover_target=failover_target)
+    LOGGER.info("checking failover status")
+    for count in range(4):
```

Comment:
Why are we checking 4 times? What if it takes longer than 2 minutes?

Reply:
failover generally  completes in  within 30 seconds.  I am checking here it for 2 mins.